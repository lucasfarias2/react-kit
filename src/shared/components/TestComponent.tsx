const TestComponent = () => {
  return <div data-testid="testComponent">Test Component Text</div>;
};

export default TestComponent;
