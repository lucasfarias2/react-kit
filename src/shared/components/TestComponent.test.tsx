import { render } from '@testing-library/react';
import { expect, test } from 'vitest';
import TestComponent from './TestComponent';

test('Renders component OK', async () => {
  const test = render(<TestComponent />);

  const testComponent = await test.findByTestId('testComponent');
  expect(testComponent).toBeDefined();
  expect(testComponent.textContent).toContain('Test Component Text');
  expect(test.asFragment()).toMatchSnapshot();
  test.unmount();
});

test('Renders component OK with snapshot', async () => {
  const test = render(<TestComponent />);
  expect(test.asFragment()).toMatchSnapshot();
  test.unmount();
});

test('Renders component OK with snapshot inline', async () => {
  const test = render(<TestComponent />);
  expect(test.asFragment()).toMatchInlineSnapshot(`
    <DocumentFragment>
      <div
        data-testid="testComponent"
      >
        Test Component Text
      </div>
    </DocumentFragment>
  `);
  test.unmount();
});
