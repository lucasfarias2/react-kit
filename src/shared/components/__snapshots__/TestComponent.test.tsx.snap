// Vitest Snapshot v1

exports[`Renders component OK 1`] = `
<DocumentFragment>
  <div
    data-testid="testComponent"
  >
    Test Component Text
  </div>
</DocumentFragment>
`;

exports[`Renders component OK with snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="testComponent"
  >
    Test Component Text
  </div>
</DocumentFragment>
`;
